Fix errors in FAQ question about List.filled

stereotype441 opened the pull request:
There was a syntax error (`List?<Foo?>`).  Also the example talked about initializing a list with the same value when in fact the "bad" example was initializing each element of the list with a different value.  This change fixes the syntax error and splits the one example into two (one using `List.filled`, for when there really is the same value, and the other using `List.generate`).

kwalrath commented on src/null-safety/faq.md:
```diff
@@ -249,20 +249,33 @@ use the [`filled`](https://api.dart.dev/stable/dart-core/List/List.filled.html)
-readable.
+If you are setting the elements of the list via an index, or you are populating
+each element of the list with a distinct value, you should instead use the
+[`generate`](https://api.dart.dev/stable/dart-core/List/List.generate.html) constructor to build the list.
```
Let's keep the line length <= 80, where possible. (Normally we'd also use a macro for the API doc link, but since this doc is a copy of internal doc... nope.)

```suggestion
[`generate`](https://api.dart.dev/stable/dart-core/List/List.generate.html)
constructor to build the list.
```

kwalrath reviewed (approved):
Thanks, @stereotype441!

natebosch commented on src/null-safety/faq.md:
```diff
@@ -249,20 +249,33 @@ use the [`filled`](https://api.dart.dev/stable/dart-core/List/List.filled.html)
+{:.bad}
+{% prettify dart tag=pre+code %}
+_jellyPoints = List<Vec2D?>(jellyMax + 1);
+for (var i = 0; i <= jellyMax; i++) {
```
[nit] The rewrite I would do is

```dart
_jellyPoints = [
  for (var i = 0; i <= jellyMax; i++)
    Vec2D(),
 ];
 ```
 
 We could make this slightly more realistic if we add a `growable: false` argument to `List.generate` which makes it infeasible to use the list literal. I'm not sure if doing that would add noise that detracts from the point though?

stereotype441 replied:
Honestly that's the rewrite I would have done too.  But I have the impression that `List.generate` is what we're trying to steer users to these days (and that the VM and dart2js folks have put special effort into making sure it's performant), so I figured 
I would go with that.  @leafpetersen / @munificent, do either of you have a preference for what we recommend here?

leafpetersen replied:
List.generate was optimized: https://github.com/dart-lang/sdk/issues/42283 . Not sure about the collection literal, maybe @alexmarkov , @mraleph or @rakudrama can comment here.  In general though, I'd probably steer people first and foremost towards the most maintainable/readable code (so long as perf is adequate).  So my vote is probably for the collection literal.

stereotype441 replied:
Ok, switched to a collection literal.

stereotype441 commented:
@natebosch @leafpetersen @kwalrath I believe I've adequately addressed review comments, but I don't have the necessary permissions to land this PR.  Can one of you land it?
